pkg/sql/logictest/testdata/logic_test/cast

```diff
@@ -1,3 +1,88 @@
+# Tests for assignment casts.
+subtest assignment_casts
+
+statement ok
+CREATE TABLE assn_cast (
+  c CHAR,
+  vc VARCHAR(1),
+  qc "char",
+  b BIT,
+  i INT,
+  t timestamp
+)
+
+statement ok
+INSERT INTO assn_cast(c) VALUES ('a')
+
+statement error value too long for type CHAR
+INSERT INTO assn_cast(c) VALUES ('abc')
+
+query T
+INSERT INTO assn_cast(c) VALUES (1) RETURNING c
+----
+1
+
+statement error value too long for type CHAR
+INSERT INTO assn_cast(c) VALUES (123)
+
+statement ok
+INSERT INTO assn_cast(vc) VALUES ('a')
+
+statement error value too long for type VARCHAR\(1\)
+INSERT INTO assn_cast(vc) VALUES ('abc')
+
+query T
+INSERT INTO assn_cast(c) VALUES (1) RETURNING c
+----
+1
+
+statement error value too long for type CHAR
+INSERT INTO assn_cast(c) VALUES (123)
+
+statement ok
+INSERT INTO assn_cast(qc) VALUES ('a')
+
+query T
+INSERT INTO assn_cast(qc) VALUES ('abc') RETURNING qc
+----
+a
+
+# Note: This statement fails in Postgres because the default integer type is an
+# INT4, and the INT4 -> "char" cast is explicit. Our default integer type
+# is an INT8 and INT8 -> "char" is an assignment cast.
+#
+# TODO(mgartner): This should return '{'.
```

Comment:
up to you: i think rather than giving the TODO for yourself for these, file an issue and use TODO(#issue_num). that way you can abandon the baby ;) (and maybe pawn it off to uhhh an unsuspecting team)

Reply:
I created #70771, but I might try to get to it before you do 😉